ckan/lib/dictization/model_save.py

```diff
@@ -30,6 +32,11 @@ def resource_dict_save(res_dict, context):
     table = class_mapper(model.Resource).mapped_table
     fields = [field.name for field in table.c]
 
+    # Strip the full url for resources of type 'upload'
+    if res_dict.get('url_type') == u'upload':
+        url = res_dict.get('url')
+        res_dict[u'url'] = url[url.rfind(u"/")+1:]
```

Comment:
This will fail if res_dict has no url or if it's set to `None`, also `rfind` is a pretty rarely used string function. How about:
```python
if res_dict.get('url') and res_dict.get('url_type') == u'upload':
    res_dict['url'] = res_dict['url'].rsplit('/')[-1]
```